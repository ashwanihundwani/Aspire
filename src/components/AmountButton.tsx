import React from 'react'
import { View, TouchableOpacity, Text, StyleSheet } from 'react-native'
import { AppColors, CommonString } from '../utils/constants'
import { FontStyle, avenirFont, windowWidth } from '../utils/utils'

const amountButtonWidth = windowWidth() * 0.275

interface AmountButtonProps {
    amount: string,
    onClick: () => void
}

const AmountButton: React.FC<AmountButtonProps> = (props: AmountButtonProps) => {
    return <TouchableOpacity
        onPress={() => {
            props.onClick()
        }}>
        <View style={styles.content}>
            <Text style={styles.buttonText}>{CommonString.currencySign + " " + props.amount}</Text>
        </View>
    </TouchableOpacity>
}

export default React.memo<AmountButtonProps>(AmountButton)

const styles = StyleSheet.create({

    content: {
        justifyContent: "center",
        alignItems: "center",
        height: 40,
        width: amountButtonWidth,
        backgroundColor: AppColors.lightGreenWithOpacity,
        borderRadius: 4,
    },
    buttonText: {
        position: "absolute",
        color: AppColors.lightGreen,
        fontSize: 12,
        textAlign: "center",
        fontFamily: avenirFont(FontStyle.DemiBold)
    }
})