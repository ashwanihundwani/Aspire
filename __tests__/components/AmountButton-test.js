/**
 * @format
 */

import 'react-native';
import React from 'react';
import AmountButton from '../../src/components/AmountButton';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
      .create(<AmountButton amount={"3,000"} onClick={()=> {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});