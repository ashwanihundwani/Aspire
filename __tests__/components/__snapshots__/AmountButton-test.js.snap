// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(1,209,103, 0.07)",
        "borderRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "width": 206.25000000000003,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#01D167",
          "fontFamily": "AvenirNext-DemiBold",
          "fontSize": 12,
          "position": "absolute",
          "textAlign": "center",
        }
      }
    >
      S$ 3,000
    </Text>
  </View>
</View>
`;
